test/csuite/wt6616_checkpoint_oldest_ts/main.c

```diff
@@ -0,0 +1,383 @@
+        }
+    }
+    /* NOTREACHED */
+}
+
+/*
+ * Child process creates the database and table, and then creates the worker thread to add data
+ * until it is killed by the parent.
+ */
+static void run_workload(void) WT_GCC_FUNC_DECL_ATTRIBUTE((noreturn));
+static void
+run_workload(void)
+{
+    WT_CONNECTION *conn;
+    WT_SESSION *session;
+    wt_thread_t *thr;
+    uint32_t i;
+    char envconf[512];
+
+    thr = dcalloc(2, sizeof(*thr));
+
+    if (chdir(home) != 0)
+        testutil_die(errno, "Child chdir: %s", home);
+    testutil_check(__wt_snprintf(envconf, sizeof(envconf), ENV_CONFIG));
+
+    printf("wiredtiger_open configuration: %s\n", envconf);
+    testutil_check(wiredtiger_open(NULL, NULL, envconf, &conn));
+    testutil_check(conn->open_session(conn, NULL, NULL, &session));
+
+    /* Create the table. */
+    if (log_table)
+        testutil_check(
+          session->create(session, uri, "key_format=S,value_format=u,log=(enabled=true)"));
+    else
+        testutil_check(
+          session->create(session, uri, "key_format=S,value_format=u,log=(enabled=false)"));
+    testutil_check(session->close(session, NULL));
+
+    /* The checkpoint thread is added at the end. */
+    printf("Create checkpoint thread\n");
+    testutil_check(__wt_thread_create(NULL, &thr[0], thread_ckpt_run, conn));
+    printf("Create the worker thread\n");
+    testutil_check(__wt_thread_create(NULL, &thr[1], thread_run, conn));
+
+    /* The threads never exit, so the child will just wait here until it is killed. */
+    fflush(stdout);
+    for (i = 0; i <= 2; ++i)
+        testutil_check(__wt_thread_join(NULL, &thr[i]));
+
+    /* NOTREACHED */
+    free(thr);
+    exit(EXIT_SUCCESS);
+}
+
+/*
+ * Signal handler to catch if the child died unexpectedly.
+ */
+static void
+handler(int sig)
+{
+    pid_t pid;
+
+    WT_UNUSED(sig);
+    pid = wait(NULL);
+    /* The core file will indicate why the child exited. Choose EINVAL here. */
+    testutil_die(EINVAL, "Child process %" PRIu64 " abnormally exited", (uint64_t)pid);
+}
+
+extern int __wt_optind;
+extern char *__wt_optarg;
+
+int
+main(int argc, char *argv[])
+{
+    struct sigaction sa;
+    struct stat sb;
+    WT_CONNECTION *conn;
+    WT_CURSOR *cursor;
+    WT_DECL_RET;
+    WT_RAND_STATE rnd;
+    WT_SESSION *session;
+    pid_t pid;
+    uint64_t oldest_ts, stable_ts, ts;
+    uint32_t timeout;
+    int ch, status;
+    char kname[64], statname[1024], tscfg[64];
+    char ts_string[WT_TS_HEX_STRING_SIZE];
+    const char *working_dir;
+    bool fatal, rand_time;
+
+    (void)testutil_set_progname(argv);
+
+    rand_time = false;
+    timeout = MIN_TIME;
+    working_dir = "WT_TEST.wt6616-checkpoint-oldest-ts";
+
+    while ((ch = __wt_getopt(progname, argc, argv, "h:lt:")) != EOF)
+        switch (ch) {
+        case 'h':
+            working_dir = __wt_optarg;
+            break;
+        case 'l':
+            log_table = true;
```

Comment:
If you go with both an oplog-like and collection-like table, this arg would go away.